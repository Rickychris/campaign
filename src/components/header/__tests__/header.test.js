import React from "react";
import ReactDom from "react-dom";
import { BrowserRouter } from "react-router-dom";
import { render, cleanup } from "@testing-library/react";
import Header from "../header";

describe("Header tests", () => {
  beforeEach(cleanup);
  const headerTest = (
    <BrowserRouter>
      <Header />
    </BrowserRouter>
  );
  it("renders the Header", () => {
    const div = document.createElement("div");
    ReactDom.render(headerTest, div);
  });

  it("renders the Header with Home text correctly", () => {
    const { getByTestId } = render(headerTest);
    expect(getByTestId("header")).toHaveTextContent("HOME");
  });

  it("matches snapshot", () => {
    const { getByTestId } = render(headerTest);
    expect(getByTestId("header")).toMatchSnapshot();
  });
});
