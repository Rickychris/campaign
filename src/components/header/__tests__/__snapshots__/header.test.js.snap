// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests matches snapshot 1`] = `
<div
  class="App-header-wrap"
  data-testid="header"
>
  <header
    class="App-header max-width"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        HOME
      </a>
    </div>
    <nav
      class="header-nav"
    >
      <p>
        About
      </p>
      <p>
        Contact Us
      </p>
    </nav>
  </header>
</div>
`;
